File: endpoints/photos.py
from silence.decorators import endpoint

@endpoint(
    route="/photos",
    method="GET",
    sql="SELECT * FROM PhotosWithUsersAndValoration WHERE visibility = 'Public' ORDER BY date DESC"
)
def get_all():
    pass

###############################################################################

@endpoint(
    route="/photos/all/$userId",
    method="GET",
    sql="SELECT * FROM PhotosWithUsersAndValoration WHERE visibility = 'Public' OR userId = $userId ORDER BY date DESC"
)
def get_all_by_id():
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="GET",
    sql="SELECT * FROM PhotosWithUsersAndValoration WHERE photoId = $photoId"
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/profile/public/$userId/profile",
    method="GET",
    sql="SELECT * FROM PhotosWithUsersAndValoration WHERE visibility = 'Public' AND userId = $userId ORDER BY date DESC"
)
def get_by_userId():
    pass

###############################################################################

@endpoint(
route="/profile/selfprofile/$userId/profile",
    method="GET",
    sql="SELECT * FROM PhotosWithUsersAndValoration WHERE userId = $userId ORDER BY date DESC"
)
def get_all_by_userId():
    pass

###############################################################################

@endpoint(
    route="/photos",
    method="POST",
    sql="INSERT INTO Photos (title, description, url, visibility, userId) VALUES ($title, $description, $url, $visibility, $userId)",
    description="Creates a new photo",
    auth_required=True,
)
def create(title, description, url, visibility, userId):
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="PUT",
    sql="UPDATE Photos SET title = $title, description = $description, url = $url, visibility = $visibility WHERE photoId = $photoId",
    description="Updates an existing photo",
    auth_required=True,
)
def update(title, description, url, visibility):
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="DELETE",
    sql="DELETE FROM Photos WHERE photoId = $photoId",
    description="Removes a photo",
    auth_required=True,
)
def delete():
    pass

File: docs/js/api/photos.js
"use strict";

import { BASE_URL, requestOptions } from "/js/api/common.js";

const photosAPI = {
    getAll: function () {
        return new Promise(function(resolve, reject) {
            axios.get(BASE_URL + "/photos", requestOptions)
                .then(response => resolve(response.data))
                .catch(err => reject(err.response.data.message));
        });
    },

    getAllPrivPub: function (userId) {
        return new Promise(function(resolve, reject) {
            axios.get(BASE_URL + "/photos/all/" + userId, requestOptions)
                .then(response => resolve(response.data))
                .catch(err => reject(err.response.data.message));
        });
    },

    getById: function(photoId) {
        return new Promise(function(resolve, reject){
            axios.get(BASE_URL + "/photos/" + photoId, requestOptions)
                .then(response => resolve(response.data[0]))
                .catch(err => reject(err.response.data.message));
        });
    },

    getAllUserId: function (userId) {
        return new Promise(function(resolve, reject){
            axios.get(BASE_URL + "/profile/public/" + userId + "/profile", requestOptions)
                .then(response => resolve(response.data))
                .catch(err => reject(err.response.data.message));
        });
    },

    getAllPubPrivUserId: function (userId) {
        return new Promise(function(resolve, reject){
            axios.get(BASE_URL + "/profile/selfprofile/" + userId + "/profile", requestOptions)
                .then(response => resolve(response.data))
                .catch(err => reject(err.response.data.message));
        });
    },

    create: function(formData) {
        return new Promise(function(resolve, reject) {
            axios.post(BASE_URL + "/photos", formData, requestOptions)
                .then(response => resolve(response.data))
                .catch(err => reject(err.response.data.message));
        });
    },

    update: function(photoId, formData) {
        return new Promise(function(resolve, reject) {
            axios.put(BASE_URL + "/photos/" + photoId, formData, requestOptions)
                .then(response => resolve(response.data))
                .catch(err => reject(err.response.data.message));
        });
    },

    delete: function(photoId) {
        return new Promise(function(resolve, reject) {
            axios.delete(BASE_URL + "/photos/" + photoId, requestOptions)
                .then(response => resolve(response.data))
                .catch(err => reject(err.response.data.message));
        });
    },

};

export { photosAPI };